crates/cairo-lint-core/src/fix.rs

```diff
@@ -153,6 +157,43 @@ impl Fixer {
         node.get_text(db).replace("break ();", "break;").to_string()
     }
 
+    pub fn fix_bool_comparison(&self, db: &dyn SyntaxGroup, node: ExprBinary) -> String {
+        let lhs = node.lhs(db).as_syntax_node().get_text(db);
+        let rhs = node.rhs(db).as_syntax_node().get_text(db);
+        let op_kind = node.op(db).as_syntax_node().kind(db);
+
+        let fixed_text = match (lhs.as_str(), rhs.as_str(), op_kind) {
+            ("true", _, SyntaxKind::TokenEqEq) => rhs.to_string(),
+            ("false", _, SyntaxKind::TokenEqEq) => format!("!{}", rhs),
+            ("true", _, SyntaxKind::TokenNeq) => format!("!{}", rhs),
+            ("false", _, SyntaxKind::TokenNeq) => rhs.to_string(),
+            ("true", _, SyntaxKind::TokenLT) => "false".to_string(),
+            ("true", _, SyntaxKind::TokenLE) => "false".to_string(),
+            ("true", _, SyntaxKind::TokenGT) => "false".to_string(),
+            ("true", _, SyntaxKind::TokenGE) => "false".to_string(),
+            ("false", _, SyntaxKind::TokenLT) => rhs.to_string(),
+            ("false", _, SyntaxKind::TokenLE) => rhs.to_string(),
+            ("false", _, SyntaxKind::TokenGT) => rhs.to_string(),
+            ("false", _, SyntaxKind::TokenGE) => rhs.to_string(),
+            (_, "true", SyntaxKind::TokenEqEq) => lhs.to_string(),
+            (_, "false", SyntaxKind::TokenEqEq) => format!("!{}", lhs),
+            (_, "true", SyntaxKind::TokenNeq) => format!("!{}", lhs),
+            (_, "false", SyntaxKind::TokenNeq) => lhs.to_string(),
+            (_, "true", SyntaxKind::TokenLT) => "false".to_string(),
+            (_, "true", SyntaxKind::TokenLE) => "false".to_string(),
+            (_, "true", SyntaxKind::TokenGT) => "false".to_string(),
+            (_, "true", SyntaxKind::TokenGE) => "false".to_string(),
+            (_, "false", SyntaxKind::TokenLT) => lhs.to_string(),
+            (_, "false", SyntaxKind::TokenLE) => lhs.to_string(),
+            (_, "false", SyntaxKind::TokenGT) => lhs.to_string(),
+            (_, "false", SyntaxKind::TokenGE) => lhs.to_string(),
+            // Handle other cases
+            _ => node.as_syntax_node().get_text(db).to_string(),
+        };
```

Comment:
maybe the match can be extracted into a function and call this function for `lhs` and `rhs`. Also I think we shouldn't lint for `<` `<=` `>=` and `>` as these implementations are not provided in the corelib

Reply:
done!